daemons/ipa-sam/ipa_sam.c

```diff
@@ -2239,7 +2239,7 @@ static bool handle_cross_realm_princs(struct ipasam_private *ipasam_state,
 				 * POSIX and SMB identities associated with our domain but we associate
 				 * them with the trust domain object itself */
 				failed += !set_krb_princ(ipasam_state, tmp_ctx,
-							 princ_l_tdo, l_tdo_alias,
+							 l_tdo_alias, princ_l_tdo,
```

Comment:
should we adjust the comment above: 

				/* Second: <REMOTE FLAT NAME>$@<OUR REALM>, enabled by default

				/* Second: <krbtgt/REMOTE FLAT NAME>$@<OUR REALM>, enabled by default